### PR TITLE
Bug 1899128: Show info message only when vmi not controlled

### PR DESCRIPTION
**Pull request by metalice**
New component and login to choose when to show

Signed-off-by: metalice <mschatzm@redhat.com>

### Fixes:
https://bugzilla.redhat.com/show_bug.cgi?id=1899128
 
### Analysis / Root cause:
Info message didnt checked if VMI controlled by a VM before presenting message

### Solution Description:
Added a check to to decide when to present the message according to owners.

### Screen shots / Gifs for design review:
bz

### CODE REVIEW

**Comment by metalice** [comment]
please review @yaacov @glekner @rawagner :)

File: frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { HintBlock, ExternalLink } from '@console/internal/components/utils';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { VirtualMachineInstanceModel, VirtualMachineModel } from '../../models/index';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+
+const InfoMessageHintBlock: React.FC<InfoMessageHintBlockProps> = ({ name, namespace }) => {
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx:8]:
```suggestion
export const VMIDetailsPageInfoMessage: React.FC<InfoMessageHintBlockProps> = ({ name, namespace }) => {
```

File: frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  const { t } = useTranslation();
+  const [resource] = useK8sWatchResource<K8sResourceCommon>({
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx:10]:
```suggestion
  const [vmi] = useK8sWatchResource<VMIKind>({
```

File: frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    kind: VirtualMachineInstanceModel.kind,
+    name,
+    namespace,
+  });
+  const isOwnedByVM = !!resource?.metadata?.ownerReferences.find(
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx:15]:
```suggestion
  const isOwnedByVM = getOwnerReferences(vmi).find(
```

File: frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    )
+  );
+};
+
+type InfoMessageHintBlockProps = {
+  name: string;
+  namespace: string;
+};
+
+export default InfoMessageHintBlock;
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx:45]: hmm, `export default` is not polite, some editors will ignore it when suggesting auto imports, and users may misspell the component when importing, can we use named import, WDYT ?

**Review by yaacov — CHANGES_REQUESTED** [review]
Nice :+1: 

I know our code coverage is not that high, but we need to start somewhere :-)

please add a test:
``` ts
import * as React from 'react';
import { shallow, mount, ShallowWrapper } from 'enzyme';
import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
...

jest.mock('react-i18next', () => {
  const reactI18next = require.requireActual('react-i18next');
  return {
    ...reactI18next,
    useTranslation: () => ({ t: (key) => key }),
  };
});

jest.mock('@console/internal/components/utils/k8s-watch-hook', () => ({
  useK8sWatchResource: jest.fn(),
}));

describe('VMIDetailsPageInfoMessage', () => {
  it('should ...', () => {
    ...
  });
```

File: frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx (hunk 4 above)
Comment (rawagner) [reply frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx:45]: Actually its a convention in console to use camel case for component files (VMIDetailsPageInfoMessage.tsx) and use default export. Its just kubevirt plugin that didnt align yet :) I would keep the default export as is

Comment (metalice) [reply frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx:45]: Im going with convention, i agree with @rawagner.
 not related to this pr but same goes for nested components btw which is heavily used in kubevirt :)

Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx:45]: +1 from me to use console convention :+1:

File: frontend/packages/kubevirt-plugin/src/components/info-messages/__tests__/mock_data.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    {
+      apiVersion: 'kubevirt.io/v1alpha3',
+      kind: 'VirtualMachineInstance',
+      metadata: {
+        ownerReferences: [
+          {
+            apiVersion: 'kubevirt.io/v1alpha3',
+            blockOwnerDeletion: true,
+            controller: true,
+            kind: 'VirtualMachine',
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/info-messages/__tests__/mock_data.ts:14]: it's `testWithoutVMOwner` the owner is `VirtualMachine`, what am i missing ?

File: frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { HintBlock, ExternalLink } from '@console/internal/components/utils';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { getOwnerReferences } from '@console/shared/src';
+import { VirtualMachineInstanceModel, VirtualMachineModel } from '../../models/index';
+import { VMIKind } from '../../types';
+
+export const InfoMessageHintBlock: React.FC<InfoMessageHintBlockProps> = ({ name, namespace }) => {
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/info-messages/vmi-details-page-info-message.tsx:9]: name should fit the file name

File: frontend/packages/kubevirt-plugin/src/components/vms/vmi-details-page.tsx [hunk 7]
```diff
@@ -22,11 +22,10 @@ import { VMConsoleFirehose } from './vm-console';
 import { VMDetailsFirehose } from './vm-details';
 import { vmiMenuActionsCreator } from './menu-actions';
 import { VMDashboard } from './vm-dashboard';
+import InfoMessageHintBlock from '../info-messages/vmi-details-page-info-message';
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/vms/vmi-details-page.tsx:25]: wooo ...

Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/vms/vmi-details-page.tsx:25]: missed the typo here, cancel the lgtm

p.s. we also need to change the file names to camel-case (e.g. `VMIDetailsPageInfoMessage.tsx` and  `VMIDetailsPageInfoMessage.spec.tsx`)

**Pull request merged**